clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -335,38 +335,28 @@ void Sema::CheckDeprecatedSYCLAttributeSpelling(const ParsedAttr &A,
     return;
   }
 
-  // Diagnose SYCL 2017 spellings in later SYCL modes.
-  if (LangOpts.getSYCLVersion() > LangOptions::SYCL_2017) {
-    // All attributes in the cl vendor namespace are deprecated in favor of a
-    // name in the sycl namespace as of SYCL 2020.
-    if (A.hasScope() && A.getScopeName()->isStr("cl")) {
-      DiagnoseDeprecatedAttribute(A, "sycl", NewName);
-      return;
-    }
-
-    // All GNU-style spellings are deprecated in favor of a C++-style spelling.
-    if (A.getSyntax() == ParsedAttr::AS_GNU) {
-      // Note: we cannot suggest an automatic fix-it because GNU-style
-      // spellings can appear in locations that are not valid for a C++-style
-      // spelling, and the attribute could be part of an attribute list within
-      // a single __attribute__ specifier. Just tell the user it's deprecated
-      // manually.
-      //
-      // This currently assumes that the GNU-style spelling is the same as the
-      // SYCL 2020 spelling (sans the vendor namespace).
-      Diag(A.getLoc(), diag::warn_attribute_spelling_deprecated)
-          << "'" + A.getNormalizedFullName() + "'";
-      Diag(A.getLoc(), diag::note_spelling_suggestion)
-          << "'[[sycl::" + A.getNormalizedFullName() + "]]'";
-      return;
-    }
+  // Diagnose SYCL 2017 spellings.
```

Comment:
We can probably drop support for that as well. It will just be a generic "unknown attribute" warning